controller/watch/watcher.go

```diff
@@ -249,48 +249,65 @@ func (w *Watcher) Watch(t resourceType, ns string, wcb watchCallback) error {
 }
 
 func deploymentWatcher(di deployments.DeploymentInterface, wcb watchCallback) {
-	for {
-		log.Println("List all existing Deployments")
-		// First do List on the resource to bring things up to date.
-		l, err := di.List(api.ListOptions{})
-		for _, d := range l.Items {
-			log.Printf("Found Deployment name: %s\n", d.Name)
-			event := watch.Event{
-				Type:   watch.Added,
-				Object: &d,
+	// 'listChan' is our signal to know if we should re-get the full list
+	// of resources.
+	listChan := make(chan bool, 1)
+
+	go func() {
```

Comment:
For the test I don't think the parallel processing is an issue but I'd like to know about this issue in general.